proto/regen/ecocredit/v1alpha1/genesis.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+package regen.ecocredit.v1alpha1;
+
+import "regen/ecocredit/v1alpha1/types.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/regen-network/regen-ledger/x/ecocredit";
```

Comment:
I was posting similar comment to NFT: how about putting genesis into a separate package, eg `x/ecocredit/genesis` - it has different domain than RPC messages. Or even we can have two packages:
* x/ecocredit/  -- for all things related to RPC
* x/ecocredit/DAL -- for all things related to storage (data access layer)

Reply:
I agree with this separation in principle, but don't want to block this PR on it. adding a new versioning scheme outside of the one for just modules (which is the current setup) is something that I'd like to make sure is the direction we want to go in across the SDK before doing it just on our own modules.

We will already need to be doing a version upgrade for ecocredit module  proto files in regen ledger v3 as we expect the API to change possibly quite a bit. So seems fine IMO to not block the v2.0 upgrade on this question.